src/main/scala/com/lightbend/rp/sbtreactiveapp/Endpoint.scala

```diff
@@ -18,8 +18,33 @@ package com.lightbend.rp.sbtreactiveapp
 
 import scala.collection.immutable.Seq
 
-case class Endpoint(protocol: String, port: Int, acls: Seq[Acl])
+sealed trait Endpoint {
+  def name: String
+  def port: Int
+  def protocol: String
+  def version: Option[Version]
+}
+
+case class HttpEndpoint(name: String, port: Int, ingress: Seq[Ingress], version: Option[Version] = Some(MajorVersion)) extends Endpoint {
+  val protocol: String = "http"
+}
+
+object HttpEndpoint {
+  def apply(name: String, port: Int, ingress: Ingress*): HttpEndpoint = new HttpEndpoint(name, port, ingress.toVector)
+}
+
+case class TcpEndpoint(name: String, port: Int, ingress: Seq[PortIngress], version: Option[Version] = Some(MajorVersion)) extends Endpoint {
```

Comment:
Please find the proposed PR which describes all of above:

https://github.com/longshorej/sbt-reactive-app/pull/1

If you feel this makes sense, please feel free to merge them. I have made sure `sbt test` and `sbt scripted` passes before I opened the PR.

Otherwise, let's discuss when we have our next standup!

Reply:
I changed the types to something a bit different based on our discussion last week.